indico/modules/events/registration/controllers/display.py

```diff
@@ -392,3 +395,20 @@ def _check_access(self):
     def _process(self):
         filename = secure_filename(f'{self.event.title}-Ticket.pdf', 'ticket.pdf')
         return send_file(filename, generate_ticket(self.registration), 'application/pdf')
+
+
+class RHRegistrationAvatar(RH):
+    """Display a standard avatar for a registration based on the full name."""
+
+    def _process_args(self):
+        self.registration = (Registration
+                             .find(Registration.id == request.view_args['registration_id'],
+                                   ~Registration.is_deleted,
+                                   ~RegistrationForm.is_deleted).one())
+
+    def _process(self):
+        first_name = self.registration.full_name[0].upper() if self.registration.full_name else ''
+        avatar = render_template('users/avatar.svg', bg_color=get_color_for_username(self.registration.full_name),
+                                 text=first_name)
+        return send_file('avatar.svg', BytesIO(avatar.encode()), mimetype='image/svg+xml',
+                         no_cache=False, inline=True, safe=False, cache_timeout=(86400*7))
```

Comment:
What about adding a `send_default_avatar(name)` util to `.../users/util.py` containing this logic? Then you could just `return send_avatar(self.registration.first_name)` in here and do something similar in the other places where we return this default avatar.

Reply:
And together with that, we could also move the `RHProfilePictureDisplay` logic into a `send_avatar(user)`, which then uses  either send_default_avatar or the usual code to serve the custom one.

That allows you to simplify the code in here to something like this:

```python
if self.registration.user:
    return send_avatar(self.registration.user) 
return send_default_avatar(self.registration.first_name)
```